core/src/main/java/org/jsmart/zerocode/core/httpclient/oauth2/OAuth2Impl.java

```diff
@@ -0,0 +1,118 @@
+package org.jsmart.zerocode.core.httpclient.oauth2;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TimerTask;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.client.utils.URLEncodedUtils;
+import org.json.JSONObject;
+import org.json.JSONTokener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author santhoshTpixler
+ *
+ */
+
+/*
+ * Note: This implementation supports the OAuth2.0 with refresh_token
+ * 
+ * Reference: https://tools.ietf.org/html/rfc6749#page-11
+ */
+public class OAuth2Impl extends TimerTask {
+	private String clienId;
+	private String clientSecret;
+	private String refreshToken;
+	private String accessTokenURL;
+	private String grant_type;
+
+	private String accessToken;
+	private static final Logger LOGGER = LoggerFactory.getLogger(OAuth2Impl.class);
+
+	public OAuth2Impl(String clientId, String clientSecret, String refreshToken, String accountsUrl, String grant_type) {
+		this.clienId = clientId;
+		this.clientSecret = clientSecret;
+		this.refreshToken = refreshToken;
+		this.accessTokenURL = accountsUrl;
+		this.grant_type = grant_type;
+	}
+
+	public OAuth2Impl(String clientId, String clientSecret, String accountsURL, String grantType) {
+		this.clienId = clientId;
+		this.clientSecret = clientSecret;
+		this.accessTokenURL = accountsURL;
```

Comment:
> this.accessTokenURL = accountsUrl;

Can you please make the property same as "accessTokenURL" instead of ~~accountsUrl~~ ?

Fix:
```
this.accessTokenURL = accessTokenURL;
```